Client/multiplayer_sa/CMultiplayerSA.cpp

```diff
@@ -7046,3 +7052,65 @@ void _declspec(naked) HOOK_CAutomobile__dmgDrawCarCollidingParticles()
         jmp RETURN_CAutomobile__dmgDrawCarCollidingParticles
     }
 }
+
+// Disable camera collisions for projectiles and vehicle parts
+const DWORD RETURN_CCollision__CheckCameraCollisionObjects = 0x41AB98;
+const DWORD RETURN_CCollision__CheckCameraCollisionObjects_2 = 0x41AC26;
+CEntitySAInterface* pLastCameraCollisionCheckObjectEntity = NULL;
+
+static bool CanEntityCollideWithCamera()
+{
+    switch (pLastCameraCollisionCheckObjectEntity->m_nModelIndex)
+    {
+        // projectiles
+        case 342: // grenade
+        case 343: // teargas
+        case 344: // molotov
+        case 363: // satchel
+
+        // vehicle parts
+        case 374: // car_door
+        case 375: // car_bumper
+        case 376: // car_panel
+        case 377: // car_bonnet
+        case 378: // car_boot
+        case 379: // car_wheel
+            return false;
+    }
+
+    return true;
+}
+
+void _declspec(naked) HOOK_CCollision__CheckCameraCollisionObjects()
+{
+    _asm
+    {
+        // Restore instructions replaced by hook
+        jz      loc_41AC26
```

Comment:
`jz RETURN_CCollision__CheckCameraCollisionObjects_2`?

Reply:
But it can jump to label only?

Reply:
I have tweaked it a bit.